static/src/javascripts/bootstraps/enhanced/newsletters.js

```diff
@@ -99,11 +106,25 @@ const createSubscriptionFormEventHandlers = (buttonEl) => {
         event.preventDefault();
         const form = buttonEl.form;
         if (validate(form)) {
-            submitForm(form, buttonEl);
+            if (window.guardian.config.switches.emailSignupRecaptcha) {
+                showCaptcha(form, () => submitForm(form, buttonEl))
+            } else {
+                submitForm(form, buttonEl);
+            }
         }
     });
 };
 
+const showCaptcha = (form, callback) => {
+    const captchaContainer = $('.grecaptcha_container', form).get(0)
```

Comment:
We only ever expect 1 of the forms on the page? Interesting point though: in newsletterContent we define grecaptcha_container as a class name, but in the recaptcha twirl template we define it as an id. Is there an intention behind this difference?

Reply:
On this page (`applications/app/views/signup/newsletterContent.scala.html`), we should have as many captcha containers (`.grecaptcha_container`) as we have signup forms. The reasons for this are:

* We want the captcha to display visually in the right place, and so it is important that the container is next to the button that triggers the captcha
* Each captcha container should be a child of the form. This is so the field (`g-recaptcha-response`) is available when submitting, and so that each submit has a fresh token. If a token has already been used, it will fail Google's validation step

Regarding id/class - there was no intention other than to avoid having multiple elements on the page with the same ID. I will update the other implementation to also use a class name.